File: frontend/src/types/deliveryOrder.ts
export type DeliveryOrderStatus = 'draft' | 'picking' | 'shipped' | 'delivered' | 'cancelled';

export interface DeliveryOrder {
    id: number;
    do_number: string;
    warehouse_id: number;
    warehouse_name?: string;
    sales_channel_id?: number;
    sales_channel_name?: string;
    customer_name: string;
    customer_address?: string;
    delivery_date: string;
    status: DeliveryOrderStatus;
    is_posted: boolean;
    posted_by?: number;
    posted_by_name?: string;
    posted_at?: string;
    shipping_method?: string;
    tracking_number?: string;
    notes?: string;
    created_at: string;
    created_by_name?: string;
    updated_at: string;
    items?: DeliveryOrderItem[];
}

export interface DeliveryOrderItem {
    id: number;
    finished_product_id: number;
    product_name?: string;
    product_sku?: string;
    warehouse_location_id?: number;
    location_code?: string;
    batch_number?: string;
    lot_number?: string;
    expiry_date?: string;
    quantity: number;
    unit_cost?: number;
    notes?: string;
}

export interface CreateDeliveryOrderRequest {
    warehouse_id: number;
    sales_channel_id?: number;
    customer_name: string;
    customer_address?: string;
    delivery_date: string; // YYYY-MM-DD
    shipping_method?: string;
    notes?: string;
    items: CreateDeliveryOrderItem[];
}

export interface CreateDeliveryOrderItem {
    finished_product_id: number;
    quantity: number;
    warehouse_location_id?: number;
    batch_number?: string;
    lot_number?: string;
    notes?: string;
}

export interface DeliveryOrderFilter {
    warehouse_id?: number;
    sales_channel_id?: number;
    status?: string;
    do_number?: string;
    customer_name?: string;
    offset?: number;
    limit?: number;
}

export interface UpdateDeliveryOrderRequest {
    sales_channel_id?: number;
    customer_name?: string;
    customer_address?: string;
    delivery_date?: string;
    shipping_method?: string;
    tracking_number?: string;
    notes?: string;
    items?: CreateDeliveryOrderItem[];
}

export interface ShipDeliveryOrderRequest {
    tracking_number?: string;
    notes?: string;
}

File: backend/internal/dto/delivery_order.go
package dto

type CreateDeliveryOrderRequest struct {
	WarehouseID     uint                     `json:"warehouse_id" binding:"required"`
	CustomerName    string                   `json:"customer_name" binding:"required"`
	CustomerAddress string                   `json:"customer_address"`
	DeliveryDate    string                   `json:"delivery_date" binding:"required"` // YYYY-MM-DD
	ShippingMethod  string                   `json:"shipping_method"`
	SalesChannelID  *uint                    `json:"sales_channel_id"`
	Notes           string                   `json:"notes"`
	Items           []CreateDeliveryOrderItem `json:"items" binding:"required,gt=0"`
}

type CreateDeliveryOrderItem struct {
	FinishedProductID uint    `json:"finished_product_id" binding:"required"`
	Quantity          float64 `json:"quantity" binding:"required,gt=0"`
	WarehouseLocationID *uint  `json:"warehouse_location_id"`
	BatchNumber       string  `json:"batch_number"`
	LotNumber         string  `json:"lot_number"`
	Notes             string  `json:"notes"`
}

type DeliveryOrderFilterRequest struct {
	WarehouseID  uint   `form:"warehouse_id"`
	Status       string `form:"status"`
	DONumber       string `form:"do_number"`
	CustomerName   string `form:"customer_name"`
	SalesChannelID uint   `form:"sales_channel_id"`
	Offset         int    `form:"offset,default=0"`
	Limit        int    `form:"limit,default=20"`
}

type UpdateDeliveryOrderRequest struct {
	CustomerName    string                   `json:"customer_name"`
	CustomerAddress string                   `json:"customer_address"`
	DeliveryDate    string                   `json:"delivery_date"`
	ShippingMethod  string                   `json:"shipping_method"`
	TrackingNumber  string                   `json:"tracking_number"`
	SalesChannelID  *uint                    `json:"sales_channel_id"`
	Notes           string                   `json:"notes"`
	Items           []CreateDeliveryOrderItem `json:"items"`
}

type ShipDeliveryOrderRequest struct {
	TrackingNumber string `json:"tracking_number"`
	Notes          string `json:"notes"`
}
